core/app/app.go

```diff
@@ -177,53 +147,23 @@ func (c *Connector) Token() net.Token {
 	return c.token
 }
 
-func (c *Connector) threadToApp(con Connection, wg *sync.WaitGroup) {
-	defer c.goRoutines.Done()
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	sub, err := con(ctx, c.threadID)
-	if err != nil {
-		log.Fatalf("error getting thread subscription: %v", err)
-	}
-	wg.Done()
-	for {
-		select {
-		case <-c.closeChan:
-			log.Debugf("closing thread-to-app flow on thread %s", c.threadID)
-			return
-		case rec, ok := <-sub:
-			if !ok {
-				log.Debug("notification channel closed, not listening to external changes anymore")
-				return
-			}
-			if err = c.app.HandleNetRecord(rec, c.threadKey, c.logID, fetchEventTimeout); err != nil {
-				log.Fatal(err)
-			}
-		}
-	}
+// CreateNetRecord calls net.CreateRecord while supplying thread ID and API token.
+func (c *Connector) CreateNetRecord(ctx context.Context, body format.Node, token thread.Token) (net.ThreadRecord, error) {
```

Comment:
These connector methods are mostly just helpers that add connector params like thread ID and API token to the underlying methods.